engine/src/srvoutput.cpp

```diff
@@ -509,11 +517,11 @@ bool MCServerSetCookie(MCStringRef p_name, MCStringRef p_value, uint32_t p_expir
 	
 	if (t_success)
 	{
-        MCservercgicookies[t_index].name = MCStringIsEmpty(p_name) ? strdup("") : strdup(MCStringGetCString(p_name));
-		MCservercgicookies[t_index].value = strdup(MCStringGetCString(*t_encoded));
-        MCservercgicookies[t_index].path = MCStringIsEmpty(p_domain) ? strdup("") : strdup(MCStringGetCString(p_path));
-        MCservercgicookies[t_index].domain = MCStringIsEmpty(p_domain) ? strdup("") : strdup(MCStringGetCString(p_domain));
-
+		// SN-2015-05-14: [[ MCservercgicookies Refactor ]] Now stores StringRefs
```

Comment:
Looking at this, it seems that t_index could cover an existing element thus we are overwriting the name, value, path, domain fields. Thus I think we should be using MCValueAssign() here to ensure the previous values are released before updating.